### PR TITLE
Add SQL specific tests for history shard

### PR DESCRIPTION
**Pull request by wxing1292**
* Add history shard tests

**What changed?**
Add SQL persistence specific tests.

**Why?**
This PR provides better test coverage and code management.

**How did you test it?**
Run tests.

**Potential risks**
Minor business logic code change.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -219,7 +219,7 @@ goimports-check:
 
 goimports:
 	@printf $(COLOR) "Run goimports..."
-	@goimports -w $(ALL_SRC)
+	@goimports -local "go.temporal.io" -w $(ALL_SRC)
```
Comment (alexshtin) [inline Makefile:222]: See #793 comments on this.

File: common/persistence/sql/sqlplugin/history.go [hunk 2]
```diff
@@ -25,13 +25,13 @@ package sqlplugin
...
-	// historyShard is the SQL persistence interface for history shards
-	historyShard interface {
+	// HistoryShard is the SQL persistence interface for history shards
+	HistoryShard interface {
 		InsertIntoShards(rows *ShardsRow) (sql.Result, error)
 		UpdateShards(row *ShardsRow) (sql.Result, error)
 		SelectFromShards(filter *ShardsFilter) (*ShardsRow, error)
-		ReadLockShards(filter *ShardsFilter) (int, error)
-		WriteLockShards(filter *ShardsFilter) (int, error)
+		ReadLockShards(filter *ShardsFilter) (int64, error)
```
Comment (alexshtin) [inline common/persistence/sql/sqlplugin/history.go:33]: I think @sergeybykov converted (wanted to convert) all `shardID` to `int32` from `int64`. Please check with him.

**Review by alexshtin — APPROVED** [review]

File: common/persistence/sql/sqlplugin/history.go (hunk 2 above)
Comment (wxing1292) [reply common/persistence/sql/sqlplugin/history.go:33]: not really, lots of shard IDs are still either `int` or `int64`
this is a short term solution. in my next PRs, all will be converted to `int32`

Comment (sergeybykov) [reply common/persistence/sql/sqlplugin/history.go:33]: I only touched those usages where they were being explicitly converted to strings. There are still `int`s and `int64`s left to clean up.

Comment (wxing1292) [reply common/persistence/sql/sqlplugin/history.go:33]: @sergeybykov i will carry out the change.

**Pull request merged**